common/djangoapps/edeos/views.py

```diff
@@ -49,11 +49,13 @@ def update_wallet(request):
                 "base_url": getattr(settings, 'EDEOS_API_URL')
             }
             response = send_edeos_api_request(**edeos_post_data)
-            if response:
+            if response and not isinstance(response, int):
                 profile = UserProfile.objects.filter(user=request.user).first()
                 if profile:
                     profile.save_profitonomy_public_key(profitonomy_public_key)
                     profile.save_wallet_name(wallet_name)
+            elif response == 422:
```

Comment:
Could use `httplib.UNPROCESSABLE_ENTITY` instead of `422`

Reply:
Agree, fixed.